### DISCUSSION TITLE
Redirect to /_ after Google OAuth

### DISCUSSION THREAD

**Discussion by andriykrefer** [pocketbase/pocketbase#4535] [Q&A] [answered]
I noticed that right after I login via google, a can briefly see a flash of 404 page, then the regular page opens correctly.

Investigating a little bit I saw in the nginx log that the 404 page is from a redirect to /_ route.

Let me explain my setup:

- pocketbase is running behind a nginx proxy at https://pb.mydomain.com
- https://pb.mydomain.com/_ is redirecting to https://mydomain.com/404 to prevent public access to the admin panel
- In the google console, the redirect callback is configured as "https://pb.mydomain.com/api/oauth2-redirect"

My question is why pocketbase is briefly redirecting to /_ ? It seems to not being used for anything, because even returning a 404 error, the login process continues correctly. Is any configuration about it? Can I disable it?

**Comment by ganigeorgiev** [comment, marked as answer]
When using the "All-in-one" OAuth2 flow we redirect to default success or failure pages (`/_/#/auth/oauth2-redirect-success` or `/_/#/auth/oauth2-redirect-failure`). These pages are rendered in the OAuth2 result popup and in case the popup cannot be closed the user will see a fallback message like: _"Auth completed. You can close this window and go back to the app."_

Note that redirecting `/_` to 404 will not prevent others making admin login attempts because the Admin UI is just a static browser application (Svelte SPA) that under the hood is sending requests to the PocketBase web APIs.

If you want to prevent others making admin auth requests, you need to block the `/api/admins*` path.

**Reply by andriykrefer** [reply]
Thanks for the response!

Maybe it's more a psychological factor, but it was bothering me to leave the admin panel open.
However, as it may interfere with the auth flow, it is better to leave it open.
Blocking /api/admins* would bee too drastic for me as I plan to access the admin api remotely in the future.
I know /_ is just a front-end but, but I had it closed because some bots lookup for login forms and interact with them, and not so much with random api routes.

Thanks again,

**Reply by j0sh** [reply]
You can use a reverse proxy to selectively block API routes to the public (as well as `/_`), while leaving them accessible via other means. I use Cloudflare Tunnel as my proxy, but nginx / caddy / etc will work as well. My admin stuff is only accessible via localhost (SSH port forwarding) but you can do similar with a private network overlay, eg Tailscale.

**Reply by bvqbao** [reply]
> You can use a reverse proxy to selectively block API routes to the public (as well as `/_`), while leaving them accessible via other means. I use Cloudflare Tunnel as my proxy, but nginx / caddy / etc will work as well. My admin stuff is only accessible via localhost (SSH port forwarding) but you can do similar with a private network overlay, eg Tailscale.

How do you block routes/paths with Cloudflare Tunnel? Don't see any option about blocking paths with Cloudflare Tunnel.

**Reply by j0sh** [reply]
> How do you block routes/paths with Cloudflare Tunnel 

Match on a path (eg, `_/`) and set the service to some other public URL
